Добавлены модели, контроллеры, валидация, логирование, тесты

Wave0no0 opened the pull request.

Wave0no0 committed:
Добавлены модели, контроллеры, валидация, логирование, тесты

Wave0no0 committed:
some fix

Wave0no0 committed:
fix

Wave0no0 committed:
test update

Wave0no0 committed:
test update + fix

Wave0no0 committed:
final fix?

Wave0no0 committed:
not final fix

Wave0no0 committed:
fix

Wave0no0 committed:
fix

Wave0no0 committed:
fix

Wave0no0 committed:
fix

Wave0no0 committed:
fix

Sla-als commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,110 @@
+            film.setId(newId);
+            films.add(film);
+            log.info("Фильм успешно добавлен: {}", film);
+            return ResponseEntity.ok(film);
```
В методе addFilm (POST "/films") используется ResponseEntity.ok(...), что возвращает статус 200 (OK) вместо 201 (Created). 

По REST-конвенциям рекомендуется при успешном создании сущности возвращать статус 201. В тестах тоже обычно ожидают isCreated().

Sla-als commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,110 @@
+    }
+
+    @PutMapping
+    public ResponseEntity<?> updateFilmWithoutId(@RequestBody Film film) {
```
В FilmController для обновления есть два метода: 
updateFilmWithoutId(@RequestBody Film film) и updateFilm(@PathVariable int id, @RequestBody Film film). Обычно в REST-приложении достаточно одного метода PUT /films/{id}, куда в пути передаётся идентификатор, а в теле — данные для обновления. Наличие двух методов может запутать пользователей API и приводит к лишним дублированиям.

Аналогично, в UserController POST/PUT возвращают не ResponseEntity<>, а просто User. Лучше придерживаться единого стиля: либо везде возвращаем сущность напрямую, либо используем ResponseEntity<>.

Sla-als commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,56 @@
+@RestController
+@RequestMapping("/users")
+@RequiredArgsConstructor // Добавляет конструктор для final полей (если они есть)
+public class UserController {
```
В UserController не используется аннотация @Slf4j, нет явных логов при добавлении или обновлении. В ТЗ сказано, что добавление и обновление сущностей следует логировать. Можно добавить хотя бы log.info() для симметрии с FilmController.

Sla-als commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,56 @@
+
+@RestController
+@RequestMapping("/users")
+@RequiredArgsConstructor // Добавляет конструктор для final полей (если они есть)
```
Я бы советовал удалить очевидные комментарии

Sla-als reviewed (commented):
Есть комментарии:

Wave0no0 committed:
Change addFilm
Remove updateFilmWithoutId
Unify return style in UserController
add logging
remove comms

Wave0no0 committed:
Checkstyle Fix

Wave0no0 committed:
Fix

Wave0no0 committed:
Fix

Wave0no0 committed:
Fix

Wave0no0 committed:
Fix

Sla-als commented:
Хорошая работа!
